best_practices/README.rst

```diff
@@ -0,0 +1,144 @@
+Python Package Compatibility Guidelines
+=======================================
+
+This document uses terminology (MUST, SHOULD, etc) from `RFC 2119`_.
+
+.. _RFC 2119: https://www.ietf.org/rfc/rfc2119.txt
+
+----------
+Background
+----------
+
+Incompatibilities between packages published on the `Python Package Index (PyPI)`_
+has been a long standing issue. Diamond dependencies are a common problem where
+a package or application has dependencies that are, themselves, dependant on
+incompatible versions of shared packages.
+has been a long standing issue. Diamond dependency (aka dependency hell) is a
+common problem that a package’s dependencies depend on different and
+incompatible versions of the shared packages.
+
+.. _Python Package Index (PyPI): https://pypi.org/
+
+Incompatibilities between packages can occur when:
+
+- A package makes breaking API changes and doesn't follow `Semantic Versioning`_
+- A package has a pinned dependency version which conflicts with other dependencies.
+- A package depends on outdated dependencies.
+- A package is depends on deprecated dependencies.
```

Comment:
```suggestion
- A package is dependent on deprecated dependencies.
```

Is this different than the above point?

Reply:
Yes depending on deprecated package could still be the latest version but it's deprecated.